subsys/net/ip/tcp.c

```diff
@@ -2343,6 +2343,11 @@ static enum net_verdict tcp_in(struct tcp *conn, struct net_pkt *pkt)
 			}
 		}
 #endif
+		if (!k_work_delayable_is_pending(
+				&conn->send_data_timer) && conn->send_data_total > 0) {
+			NET_ERR("conn: %p, Missing a subscription "
+				"of the send_data queue timer", conn);
```

Comment:
I wonder about this one, we log an error but take no action here. Is there any scenario when this would be the case other than bug somewhere else in the TCP codebase? Maybe it should be an assert?